### DISCUSSION TITLE
Trouble with 5.X Matrix and Multisite

### DISCUSSION THREAD

**Discussion by thisanimus** [craftcms/cms#15478] [Q&A] [answered]
I have a deeply-nested matrix field in a multisite (multi-lang) craft 5 install.  I'm populating the matrix field value automatically on EVENT_BEFORE_SAVE from data in an external API.  The English version of the entry has a field with the ID of the remote English data, the French version of the entry has a field with the ID of the remote French data.

I am able to create the matrix field value, but it saves to all of the language entries, not only the one that triggered the save event.  If I manually change a value in the matrix, that works as expected (the change is isolated to the siteId of the current entry.)

**Question**: Do I need to supply a siteId value to the matrix field in that save hook?  If so is there documentation on how to do this?

**Comment by brandonkelly** [comment]
Can you post what the full content model is? Including your section and Matrix fields’ Propagation Method settings?

**Reply by thisanimus** [reply, marked as answer]
Ah, got it.  thanks.  Articulating the setup helped me find the issue:


```
Course (entry type that is part of a section "courses" with propagation method: Let each entry choose which sites it should be saved to)
	- Modules (matrix field with propagation method: Save entries to all sites the owner element is saved in)
		- Module (entry type with no section association, title set to translate for each site)
			- Lessons: (matrix field with propagation method: Only save entries to the site they were created in)
```
Modules has the wrong propagation method.  it should be: Only save entries to the site they were created in

**Reply by brandonkelly** [reply]
Glad you got it sorted :)
